src/main/kotlin/com/autonomousapps/tasks/AdviceTask.kt

```diff
@@ -132,58 +137,85 @@ abstract class AdviceTask : DefaultTask() {
 
-    if (!computedAdvice.filterChange) {
-      advicePrinter.getChangeAdvice()?.let {
-        consoleReport.add("Existing dependencies which should be modified " +
-          "to be as indicated:\n$it\n")
-        didGiveAdvice = true
-      }
+    if (consoleReport.isEmpty()) {
+      chatter.chat("Looking good! No changes needed")
+    } else {
+      chatter.chat("See console report at ${adviceConsoleFile.path}")
     }
+  }
 
-    if (!computedAdvice.filterCompileOnly) {
-      advicePrinter.getCompileOnlyAdvice()?.let {
-        consoleReport.add("Dependencies which could be compile-only:\n$it\n")
-        didGiveAdvice = true
+  internal data class ConsoleReport(
+    val addToApiAdvice: Set<Advice> = emptySet(),
+    val addToImplAdvice: Set<Advice> = emptySet(),
+    val removeAdvice: Set<Advice> = emptySet(),
+    val changeToApiAdvice: Set<Advice> = emptySet(),
+    val changeToImplAdvice: Set<Advice> = emptySet(),
+    val compileOnlyDependencies: Set<Dependency> = emptySet(),
+    val unusedProcsAdvice: Set<Advice> = emptySet()) {
+
+    fun isEmpty() = addToApiAdvice.isEmpty() &&
+      removeAdvice.isEmpty() &&
+      changeToApiAdvice.isEmpty() &&
+      changeToImplAdvice.isEmpty() &&
+      compileOnlyDependencies.isEmpty() &&
+      unusedProcsAdvice.isEmpty()
+
+    companion object {
+      fun build(computedAdvice: ComputedAdvice): ConsoleReport {
+        var addToApiAdvice = emptySet<Advice>()
+        var addToImplAdvice = emptySet<Advice>()
+        var removeAdvice = emptySet<Advice>()
+        var changeToApiAdvice = emptySet<Advice>()
+        var changeToImplAdvice = emptySet<Advice>()
+        var compileOnlyDependencies = emptySet<Dependency>()
+        var unusedProcsAdvice = emptySet<Advice>()
+
+        if (!computedAdvice.filterRemove && computedAdvice.removeAdvice.isNotEmpty()) {
+          removeAdvice = computedAdvice.removeAdvice
+        }
+
+        val addAdvices = computedAdvice.addToApiAdvice + computedAdvice.addToImplAdvice
+        if (!computedAdvice.filterAdd && addAdvices.isNotEmpty()) {
+          addToApiAdvice = computedAdvice.addToApiAdvice
+          addToImplAdvice = computedAdvice.addToImplAdvice
+        }
+
+        val changeAdvices = computedAdvice.changeToApiAdvice + computedAdvice.changeToImplAdvice
+        if (!computedAdvice.filterChange && changeAdvices.isNotEmpty()) {
+          changeToApiAdvice = computedAdvice.changeToApiAdvice
+          changeToImplAdvice = computedAdvice.changeToImplAdvice
+        }
+
+        if (!computedAdvice.filterCompileOnly && computedAdvice.compileOnlyDependencies.isNotEmpty()) {
+          compileOnlyDependencies = computedAdvice.compileOnlyDependencies
+        }
+
+        // TODO add filter
+        if (computedAdvice.unusedProcsAdvice.isNotEmpty()) {
+          unusedProcsAdvice = computedAdvice.unusedProcsAdvice
+        }
+
+        return ConsoleReport(addToApiAdvice,
```

Comment:
nit: `addToApiAdvice` should be on next line, and terminal `)` should also be on its own line.

Yes I know, we need a linter.